light_node/src/configuration.rs

```diff
@@ -1089,4 +1203,50 @@ impl Environment {
             validate_cfg_identity_and_stop: args.is_present("validate-cfg-identity-and-stop"),
         }
     }
+
+    pub fn create_logger(&self) -> Logger {
+        let Environment { logging, .. } = self;
+        let drains: Vec<Arc<slog_async::Async>> = logging
+            .log
+            .iter()
+            .map(|log_target| match log_target {
+                LoggerType::TerminalLogger => Arc::new(create_terminal_logger!(logging.format)),
+                LoggerType::FileLogger => {
+                    let log_file = if let Some(path) = &logging.file {
+                        path.clone()
+                    } else {
+                        PathBuf::from("./tezedge.log")
+                    };
+                    Arc::new(create_file_logger!(logging.format, log_file))
+                }
+            })
+            .collect();
+
+        if drains.len() == 1 {
+            // if there is only one drain, return the logger
+            Logger::root(
+                drains[0].clone().filter_level(logging.level).fuse(),
+                slog::o!(),
+            )
+        } else {
+            // combine 2 or more drains into Duplicates
+
+            // need an initial value for fold, create it from the first two drains in the vector
+            let initial_value =
+                Box::new(Duplicate::new(drains[0].clone(), drains[1].clone()).fuse());
```

Comment:
@adonagy 
this will failes if drains.is_empty

Reply:
@bkontur  Fixed